src/mcmc/mcmc_misc.jl

```diff
@@ -0,0 +1,296 @@
+    df[:color_log_ratio_ri] = chain[:,4] .- chain[:,3]
+    df[:color_log_ratio_iz] = chain[:,5] .- chain[:,4]
+
+    u_samps = [constrain_pos(chain[p,6:7]) for p in 1:size(chain, 1)]
+    u_samps = transpose(hcat(u_samps...))
+
+    df[:right_ascension_deg] = u_samps[:,1]
+    df[:declination_deg]     = u_samps[:,2]
+
+    return df
+end
+
+
+function samples_to_data_frame_row(sampdf; objid="mcmc")
+    """ only for stars right now """
+    df = DataFrame()
+    df[:objid]                          = [objid]
+    df[:right_ascension_deg]            = [mean(sampdf[:right_ascension_deg])]
+    df[:declination_deg]                = [mean(sampdf[:declination_deg])]
+    df[:is_saturated]                   = [false]
+    df[:is_star]                        = [true]
+    df[:de_vaucouleurs_mixture_weight]  = [NaN]
+    df[:minor_major_axis_ratio]         = [NaN]
+    df[:half_light_radius_px]           = [NaN]
+    df[:angle_deg]                      = [NaN]
+    df[:reference_band_flux_nmgy]       = [mean(sampdf[:reference_band_flux_nmgy])]
+    df[:log_reference_band_flux_stderr] = [std(sampdf[:log_reference_band_flux])]
+    df[:color_log_ratio_ug]             = [mean(sampdf[:color_log_ratio_ug])]
+    df[:color_log_ratio_gr]             = [mean(sampdf[:color_log_ratio_gr])]
+    df[:color_log_ratio_ri]             = [mean(sampdf[:color_log_ratio_ri])]
+    df[:color_log_ratio_iz]             = [mean(sampdf[:color_log_ratio_iz])]
+    df[:color_log_ratio_ug_stderr]      = [std(sampdf[:color_log_ratio_ug])]
+    df[:color_log_ratio_gr_stderr]      = [std(sampdf[:color_log_ratio_gr])]
+    df[:color_log_ratio_ri_stderr]      = [std(sampdf[:color_log_ratio_ri])]
+    df[:color_log_ratio_iz_stderr]      = [std(sampdf[:color_log_ratio_iz])]
+ 
+    return df
+end
+
+
+# run multiple chains
+function run_multi_chain_mcmc(lnpdf::Function, th0s)
+  chains, logprobs = [], []
+  for c in 1:num_chains
+    # initialize a star state (around the existing catalog entry)
+    println("---- chain ", c, " ----")
+    th0 = init_star_params(star_state)
+    samples, lls =
+      run_mh_sampler_with_warmup(star_logpdf, th0,
+                                 num_samples,
+                                 num_warmup;
+                                 print_skip=print_skip,
+                                 keep_warmup=true,
+                                 warmup_prop_scale=warmup_prop_scale,
+                                 chain_prop_scale=chain_prop_scale)
+
+    push!(chains, samples[(num_warmup+1):end, :])
+    push!(logprobs, lls[(num_warmup+1):end])
+
+    # report PSRF after c = 3 chains
+    if c > 2
+      psrfs = MCMC.potential_scale_reduction_factor(chains)
+      println(" potential scale red factor", psrfs)
+    end
+  end
+end
+
+
+######################################################################
+# unit flux image utils --- writes unit flux images to pixel array   #
+# Very similar to Synthetic.jl functions                             #
```

Comment:
Is there duplicate code here? Could we instead modify Synthetic.jl so that those funds are general enough to work for both your use case and whatever they're doing now?